src/services/uniswap.js

```diff
@@ -311,18 +312,18 @@ export const createUniswapAllowanceTx =
     return allowanceTx;
   };
 
-export const fetchUniswapSupportedTokens = (): Promise<?string[]> => axios
-  .get(getEnv().UNISWAP_CACHED_SUBGRAPH_ASSETS_URL, defaultAxiosRequestConfig)
-  .then(({ data: responseData }: AxiosResponse) => {
-    if (!responseData) {
-      reportErrorLog('fetchUniswapSupportedTokens failed: unexpected response', { response: responseData });
+export const fetchUniswapSupportedTokens = (): Promise<?string[]> =>
+  retryOnNetworkError(() => axios.get(getEnv().UNISWAP_CACHED_SUBGRAPH_ASSETS_URL, defaultAxiosRequestConfig))
```

Comment:
Instead of wrapping every axios call, consider providing an thin axios abstraction, that would do that in a centralized form.
```
httpClient.get(...)

// Somewhere in utils
const httpClient: {
  get: (...args) => { retryOnNetworkError(() => axios.get(...args)) }
  post: etc
}
```

Reply:
I _would_ like something more elegant than a wrapper, but replacing it with an abstraction is not quite straightforward. It's supposed to be opt-in per API call, even if admittedly all GET requests I've found were ok to be wrapped. The same definitely isn't true for POST ones. A replacement also wouldn't work for SDK requests. Given that, we'd end up having some calls wrapped and some abstracted with nothing to suggest the non-standard feature. Add to that that using axios means handling axios-specific behaviour and response format and I'd rather have some light repetition if it means there's also consistency and clearer view of what exactly is being called.

I'll have think about it some more, because the opt-in part would be solved by adding a custom option to the requests, hiding axios by typing, so that leaves SDK calls. I'd most like to avoid retries being both mentioned by name and hidden in requests next to each other. Any thoughts?

Reply:
1. You're right that this probably should apply to GET calls but not POST ones, as repeated calls there could lead to unintended consequences.
2. For sure SDK requests would need to be wrapped in `retryOnNetworkErrorDirectly` but that makes only sense if they rethrow original axios error(!), any other error seems not to be retried anyway.
3. Maybe we could make retry behavior default for all GET calls, with option to opt-out. Sounds like a reasonable default. To achieve that we could add option flag to options param. Achieving non-retry behavior should probably only require specifying zero retry count without the need not-to-wrap GET call.